### ISSUE TITLE
`CREATE EXTENSION` fails linter

### ISSUE THREAD

**Issue by josephzidell** [sqlfluff/sqlfluff#709]
-- If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant. Finally, please add any labels that are related to the issue before submitting so that it can be addressed easily! --

# SQL
```sql
CREATE EXTENSION IF NOT EXISTS pg_trgm;
```

## `sqlfluff parse`
<details>

    === [ext.sql] ===
    FileSegment:                                                [0](1, 1, 1)
        StatementSegment:                                       [0](1, 1, 1)
            UnparsableSegment:                                  [0](1, 1, 1)        !! Expected: 'WITH, ... | SELECT,  !!TODO!! , *, ..., ... | VALUE | VALUES,  ( [SINGLE_QUOTE] | [NUMERIC_LITERAL] | TRUE | FALSE | + | -, NumericLiteralSegment | NULL CommaSegment INTERVAL, NumericLiteralSegment, QuotedLiteralSegment | date_part | QuotedLiteralSegment ) ,  !!TODO!!  | NonSetSelectableGrammar | INSERT, ... |  ( comment | newline | + ) | COMMIT, WORK, AND, NO, CHAIN | ROLLBACK, WorkKeywordSegment, AndKeywordSegment, NoKeywordSegment, ChainKeywordSegment | DROP, TABLE | VIEW, IF, EXISTS, identifier | [DOUBLE_QUOTE], RESTRICT | CASCADE | GRANT, ALL, PRIVILEGES | SELECT | UPDATE | INSERT,  ( identifier | [DOUBLE_QUOTE] ) , ON, TABLE, ObjectReferenceSegment | AllKeywordSegment, TABLES, IN, SCHEMA, ObjectReferenceSegment, TO, GROUP | USER | ROLE, ObjectReferenceSegment | PUBLIC, WITH, GrantKeywordSegment, OPTION | REVOKE, GrantKeywordSegment, OptionKeywordSegment, FOR, AllKeywordSegment, PrivilegesKeywordSegment | SelectKeywordSegment | UpdateKeywordSegment | InsertKeywordSegment,  ( ObjectReferenceSegment ) , OnKeywordSegment, TableKeywordSegment, ObjectReferenceSegment | AllKeywordSegment, TablesKeywordSegment, InKeywordSegment, SchemaKeywordSegment, ObjectReferenceSegment, FROM, GroupKeywordSegment | UserKeywordSegment | RoleKeywordSegment, ObjectReferenceSegment, RESTRICT | CASCADE | CREATE, OR, REPLACE, TABLE, IF, NOT, EXISTS, identifier | [DOUBLE_QUOTE],  ( SingleIdentifierGrammar, data_type_identifier,  ( bare_function | function_name,  (  <anything>  ) , WITHIN, GROUP,  (  <anything>  )  | IGNORE | RESPECT, NULLS, OVER,  (  <anything>  )  |  ( Expression_A_Grammar )  |  ( WITH, ... | SELECT,  !!TODO!! , *, ..., ... | VALUE | VALUES,  ( [SINGLE_QUOTE] | [NUMERIC_LITERAL] | TRUE | FALSE | + | -, NumericLiteralSegment | NULL CommaSegment INTERVAL, NumericLiteralSegment, QuotedLiteralSegment | date_part | QuotedLiteralSegment ) ,  !!TODO!!  | NonSetSelectableGrammar )  | SelectStatementSegment | LiteralGrammar | IntervalExpressionSegment | ObjectReferenceSegment,  !!TODO!! , ::, DatatypeSegment | CASE, ,  !!TODO!! , ELSE, , ExpressionSegment, , , END | CaseKeywordSegment, ExpressionSegment, ,  !!TODO!! , ElseKeywordSegment, , ExpressionSegment, , , EndKeywordSegment | ExistsKeywordSegment, SelectStatementSegment | + | - | NotKeywordSegment, Expression_A_Grammar,  !!TODO!!  ) ,  (  <anything>  ) ,  !!TODO!!  | CONSTRAINT, ObjectReferenceSegment, UNIQUE,  ( ObjectReferenceSegment )  | PRIMARY, KEY,  ( ObjectReferenceSegment )  | FOREIGN, KeyKeywordSegment,  ( ObjectReferenceSegment ) , REFERENCES, ObjectReferenceSegment,  ( ObjectReferenceSegment )  ) , COMMENT, QuotedLiteralSegment | AS, SelectableGrammar | LIKE, ObjectReferenceSegment | ALTER, TABLE, identifier | [DOUBLE_QUOTE], parameter, =, [SINGLE_QUOTE] | [NUMERIC_LITERAL] | TRUE | FALSE | + | -, NumericLiteralSegment | NULL | NakedIdentifierSegment | ADD | MODIFY, COLUMN, SingleIdentifierGrammar, data_type_identifier,  ( bare_function | function_name,  (  <anything>  ) , WITHIN, GROUP,  (  <anything>  )  | IGNORE | RESPECT, NULLS, OVER,  (  <anything>  )  |  ( Expression_A_Grammar )  |  ( WITH, ... | SELECT,  !!TODO!! , *, ..., ... | VALUE | VALUES,  ( LiteralGrammar CommaSegment INTERVAL, NumericLiteralSegment, QuotedLiteralSegment | date_part | QuotedLiteralSegment ) ,  !!TODO!!  | NonSetSelectableGrammar )  | SelectStatementSegment | LiteralGrammar | IntervalExpressionSegment | ObjectReferenceSegment,  !!TODO!! , ::, DatatypeSegment | CASE, ,  !!TODO!! , ELSE, , ExpressionSegment, , , END | CaseKeywordSegment, ExpressionSegment, ,  !!TODO!! , ElseKeywordSegment, , ExpressionSegment, , , EndKeywordSegment | EXISTS, SelectStatementSegment | + | - | NOT, Expression_A_Grammar,  !!TODO!!  ) ,  (  <anything>  ) ,  !!TODO!! , FIRST | AFTER, ObjectReferenceSegment |  ( ObjectReferenceSegment )  | CREATE, OR, REPLACE, VIEW, identifier | [DOUBLE_QUOTE],  ( ObjectReferenceSegment ) , AS, WITH, ... | SELECT,  !!TODO!! , *, ..., ... | VALUE | VALUES,  ( [SINGLE_QUOTE] | [NUMERIC_LITERAL] | TRUE | FALSE | + | -, NumericLiteralSegment | NULL CommaSegment INTERVAL, NumericLiteralSegment, QuotedLiteralSegment | date_part | QuotedLiteralSegment ) ,  !!TODO!!  | NonSetSelectableGrammar | DELETE, ... | UPDATE, ... | CREATE, OR, REPLACE, MODEL, IF, NOT, EXISTS, identifier | [DOUBLE_QUOTE], OPTIONS,  ( parameter, =, [SINGLE_QUOTE] | [NUMERIC_LITERAL] | TRUE | FALSE | + | -, NumericLiteralSegment | NULL |  ( QuotedLiteralSegment )  ) , AS, WITH, ... | SELECT,  !!TODO!! , *, ..., ... | VALUE | VALUES,  ( LiteralGrammar CommaSegment INTERVAL, NumericLiteralSegment, QuotedLiteralSegment | date_part | QuotedLiteralSegment ) ,  !!TODO!!  | NonSetSelectableGrammar | DROP, MODEL, IF, EXISTS, identifier | [DOUBLE_QUOTE]'
                code_RawSegment:                                [0](1, 1, 1)        'CREATE'
                whitespace_RawSegment:                          [6](1, 1, 7)        ' '
                code_RawSegment:                                [7](1, 1, 8)        'EXTENSION'
                whitespace_RawSegment:                          [16](1, 1, 17)      ' '
                code_RawSegment:                                [17](1, 1, 18)      'IF'
                whitespace_RawSegment:                          [19](1, 1, 20)      ' '
                code_RawSegment:                                [20](1, 1, 21)      'NOT'
                whitespace_RawSegment:                          [23](1, 1, 24)      ' '
                code_RawSegment:                                [24](1, 1, 25)      'EXISTS'
                whitespace_RawSegment:                          [30](1, 1, 31)      ' '
                code_RawSegment:                                [31](1, 1, 32)      'pg_trgm'
        semicolon_KeywordSegment:                               [38](1, 1, 39)      ';'
        newline_RawSegment:                                     [39](1, 1, 40)      '\n'
    
    ==== parsing violations ====
    L:   1 | P:   1 |  PRS | Found unparsable section: 'CREATE EXTENSION IF NOT EXISTS pg_trgm'

</details>

## `sqlfluff lint`
<details>

    == [ext.sql] FAIL
    L:   1 | P:   1 |  PRS | Found unparsable section: 'CREATE EXTENSION IF NOT
                           | EXISTS pg_trgm'

</details>

## `sqlfluff fix` (had to run with `--no-safety`)
<details>

    NO SAFETY: Attempting fixes for all enabled rules.
    ==== finding fixable violations ====
    ==== no fixable linting violations found ====

</details>

## Expected Behaviour
It should understand [postres extensions](https://www.postgresql.org/docs/13/external-extensions.html)

## Observed Behaviour
It fails

## Steps to Reproduce
Run the linter on the SQL command above

## Version
Include the output of `sqlfluff --version` along with your Python version

    sqlfluff, version 0.3.6
    Python 3.8.6

## Configuration
```
Include your SQLFluff configuration here
```

    [sqlfluff]
    dialect = postgres

```
